recipes/use_cases/chatbots/RAG_chatbot/vectorstore/mongodb/rag_mongodb_llama3_huggingface_open_source.ipynb

```diff
@@ -0,0 +1,1351 @@
+        "\n",
+        "The process involves preparing a dataset of arXiv papers, transforming their data for effective retrieval, setting up a MongoDB database with vector search capabilities, and using llama3 model for generating answers based on the retrieved documents.\n",
+        "\n",
+        "Key Highlights:\n",
+        "- Usage of Hugging Face open-source models and MongoDB for creating RAG pipelines.\n",
+        "- Steps include dataset preparation, database setup, data ingestion, and query processing.\n",
+        "- Detailed guidance on setting up MongoDB collections and vector search indexes.\n",
+        "- Integration with the Llama3 model from Hugging Face for answering complex queries.\n",
+        "\n",
+        "Follow the following instruction to set up a MongoDB database and enable vector search:\n",
+        "1. [Register a free Atlas account](https://account.mongodb.com/account/register?utm_campaign=devrel&utm_source=community&utm_medium=cta&utm_content=GitHub%20Cookbook&utm_term=richmond.alake)\n",
+        " or sign in to your existing Atlas account.\n",
+        "\n",
+        "2. [Follow the instructions](https://www.mongodb.com/docs/atlas/tutorial/deploy-free-tier-cluster/)\n",
+        " (select Atlas UI as the procedure)  to deploy your first cluster.\n",
+        "\n",
+        "3. Create the database: `movies`.\n",
+        "\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BPgc75ATDIvA"
+      },
+      "source": [
+        "## Import Libraries\n",
+        "\n",
+        "Import libaries into development environment"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "UUlChXcYO43e"
+      },
+      "outputs": [],
+      "source": [
+        "!pip install datasets pandas pymongo sentence_transformers\n",
+        "!pip install -U transformers\n",
+        "# Install the library below if using GPU, if using CPU, please comment out below\n",
+        "!pip install accelerate"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JZHLRiGvDSgB"
+      },
+      "source": [
+        "## Dataset Loading and Preparation\n",
+        "\n",
+        "Load the dataset from HuggingFace.\n",
+        "\n",
+        "Only using the first 100 datapoint for demo purposes."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "uRvntN3jO8j1"
+      },
+      "outputs": [],
+      "source": [
+        "# Load Dataset\n",
+        "from datasets import load_dataset\n",
+        "import pandas as pd\n",
+        "\n",
+        "# Make sure you have an Hugging Face token(HF_TOKEN) in your development environemnt before runing the code below\n",
```

Comment:
can we pls have a place holder for HF token to make sure users know how to set/ enforce it not to forget